### PR TITLE
config: change default config for Auto Routing Improved

### PR DESCRIPTION
**Pull request by kenjis**
Needs #8980

**Description**
- change the default values:
   - `Feature::$autoRoutesImproved` -> true
   - `Routing::$translateUriToCamelCase` -> true
- add new page for "Auto Routing (Improved)" in the user guide

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Screenshot 2024-06-24 at 18-40-26 Auto Routing (Improved) — CodeIgniter 4 5 2 documentation

**Review by datamweb — APPROVED** [review]
LGTM!

**Comment by kenjis** [comment]
I have made a few improvements.
Screenshot 2024-06-25 at 08-50-04 Auto Routing (Improved) — CodeIgniter 4 5 2 documentation

File: user_guide_src/source/incoming/auto_routing_improved.rst [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+
+If you did it right, you should see::
+
+    Hello World!
+
+****************************
+Examples of Controller Names
+****************************
+
+This is an valid controller name. Because ``App\Controllers\HelloWorld`` is CamelCase.
```
Comment (datamweb) [inline user_guide_src/source/incoming/auto_routing_improved.rst:183]: Honestly, I didn't like using 'This is', it made the review difficult for me.

File: user_guide_src/source/incoming/auto_routing_improved.rst [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+************
+
+The segments in the URL, in following with the Model-View-Controller approach,
+usually represent::
+
+    http://example.com/{class}/{method}/{param1}
+
+1. The first segment represents the controller **class** that should be invoked.
+2. The second segment represents the class **method** that should be called.
+3. The third, and any additional segments, represent any **parameters** that will be passed to the controller.
```
Comment (datamweb) [inline user_guide_src/source/incoming/auto_routing_improved.rst:94]: I took from this sentence that the parameters are sent to the controller. 
But the parameters are sent to the methods in the controller.

File: user_guide_src/source/incoming/auto_routing_improved.rst (hunk 1 above)
Comment (kenjis) [reply user_guide_src/source/incoming/auto_routing_improved.rst]: I changed to `The following is`.

File: user_guide_src/source/incoming/auto_routing_improved.rst (hunk 2 above)
Comment (kenjis) [reply user_guide_src/source/incoming/auto_routing_improved.rst]: Changed to `the controller method`.

**Comment by neznaika0** [comment]
I think it's worth adding a note about Filters.
Won't they be available ONLY through URIs?

**Comment by kenjis** [comment]
@neznaika0 Added.
Screenshot 2024-06-25 14 41 59

**Review by neznaika0 — APPROVED** [review]

**Review by datamweb — APPROVED** [review]
Thanks!

**Review by michalsn — APPROVED** [review]

**Pull request merged**